integration-tests/tests/src/tests/dictionary.ts

```diff
@@ -0,0 +1,315 @@
+        });
+      });
+    });
+
+    it("can store object link values using string keys", function (this: RealmContext) {
+      this.realm.write(() => {
+        const alice = this.realm.create("Person", { name: "Alice" });
+        const bob = this.realm.create("Person", { name: "Bob" });
+        // Creation
+        const item = this.realm.create<Item>("Item", {
+          dict: {
+            key1: alice,
+          },
+        });
+        expect(item.dict).deep.equals({
+          key1: alice,
+        });
+        // After assignment
+        item.dict.key1 = bob;
+        expect(item.dict).deep.equals({
+          key1: bob,
+        });
+      });
+    });
+
+    it("is spreadable", function (this: RealmContext) {
+      this.realm.write(() => {
+        const item = this.realm.create<Item>("Item", {
+          dict: { key1: "hi" },
+        });
+        expect({ ...item.dict }).deep.equals({ key1: "hi" });
+      });
+    });
+
+    it("can JSON.stringify", function (this: RealmContext) {
+      this.realm.write(() => {
+        const values: DictValues = {
+          key1: "hello",
+          key2: 1234,
+          key3: false,
+          key4: null,
+        };
+        const item = this.realm.create<Item>("Item", { dict: values });
+        const stringifiedAndParsed = JSON.parse(JSON.stringify(item.dict));
+        expect(stringifiedAndParsed).deep.equals(values);
+      });
+    });
+
+    it("can JSON.stringify via the object", function (this: RealmContext) {
+      this.realm.write(() => {
+        const values: DictValues = {
+          key1: "hello",
+          key2: 1234,
+          key3: false,
+          key4: null,
+        };
+        const item = this.realm.create<Item>("Item", { dict: values });
+        const stringifiedAndParsed = JSON.parse(JSON.stringify(item));
+        expect(stringifiedAndParsed).deep.equals({ dict: values });
+      });
+    });
+
+    // TODO: Unskip once https://github.com/realm/realm-core/issues/4805 is fixed
+    it.skip("throws a meaningful error if accessed after deletion", function (this: RealmContext) {
+      this.realm.write(() => {
+        const item = this.realm.create<Item>("Item", {});
+        const dict = item.dict;
+        this.realm.delete(item);
+        expect(() => {
+          JSON.stringify(dict);
+        }).throws("Access to invalidated Dictionary object");
+      });
+    });
+
+    it("can have values deleted", function (this: RealmContext) {
+      this.realm.write(() => {
+        const item = this.realm.create<Item>("Item", {
+          dict: { key1: "hi" },
+        });
+        expect(item.dict).deep.equals({ key1: "hi" });
+        delete item.dict.key1;
+        expect(Object.keys(item.dict)).deep.equals([]);
+        // TODO: Investigate why this blocks
+        // expect(item.dict).deep.equals({});
+      });
+    });
+  });
+
+  function describeTypedSuite({ extraSchema = [], type, goodValues, badValues, expectedError }: TypedDictionarySuite) {
```

Comment:
I was a bit confused what the `TypedDictionarySuite` was when I started reading this code from top to bottom.  I think it would be might be more clear to break these tests into their own file `typed_dictionary.ts`.